### PR TITLE
kernel: reduce arguments of FuncSHELL

### PR DESCRIPTION
**Pull request by fingolfin**
... and simplify a bunch of code using that, e.g. replace `THREAD_SESSION` by `SESSION`.

### CODE REVIEW

File: lib/error.g [hunk 1]
```diff
@@ -349,16 +348,8 @@ BIND_GLOBAL("ErrorInner", function(options, earlyMessage)
     else
         prompt := "brk> ";
     fi;
-    shellOut := "*errout*";
-    shellIn := "*errin*";
```
Comment (fingolfin) [inline lib/error.g:353]: Ah, I am stupid. Of course we must use errout/errin here. So I need to add back one argument, something like `BOOL isBreakPrompt` or so.

**Review by ChrisJefferson — APPROVED** [review]

**Comment by ChrisJefferson** [comment]
I point out this looks like it might break JupyterKernl, which calls SHELL -- anyone who wants to fix that could use NumberArgumentsFunction to tell which SHELL they are calling.

**Comment by fingolfin** [comment]
Hmm, I explicitly searched for packages using `SHELL`, and found none. I just had another look at `JupyterKernel` and still couldn't find anything. Am I missing something?

**Comment by ChrisJefferson** [comment]
False alarm, I wasn't up to date. It used to use it, but a git pull removed it.

File: src/gap.c [hunk 2]
```diff
@@ -177,7 +177,22 @@ static Obj Shell(Obj    context,
...
   STATE(ErrorLLevel) = 0;
   Int oldRecursionDepth = GetRecursionDepth();
-  
+
+  if (breakLoop) {
+    inFile = "*errin*";
+    outFile = "*errout*";
+  }
+#ifdef HPCGAP
+  if (ThreadUI) {
```
Comment (fingolfin) [inline src/gap.c:186]:
```suggestion
  else if (ThreadUI) {
```

**Pull request merged**